content/articles/searching-and-sorting-in-matlab/index.md

```diff
@@ -0,0 +1,162 @@
+
+### Prerequisites
+- [Matlab](https://www.mathworks.com/products/get-matlab.html?s_tid=gn_getml) installed.
+- Proper understanding of [matlab](https://www.section.io/engineering-education/getting-started-with-matlab/) basics.
+
+### Searching
+The main target of the search is the uniqueness of the data. For example, if you go to a bank, they ask for your bank number to get your bank account. It is made possible by search algorithms. There are two types of search algorithms. These are;
+- Sequential
+- Binary search
+
+### 1. Sequential search
+It is also known as linear search. Here, the target variable is compared to the entire vector or stop in a variable similar to the target. For example, we are given a vector and a target value to locate. First, we should identify the target value within this vector. If we find a similar variable, we give its index as the output, and if we don't, we provide an impossible index as the output. This impossible index could be something like -1. It is to show that the value is not found.
+```matlab
+function index = sequential_search(vector, target, first, last)
+found = false;                %Assumming target not found
+for n = first:last
+if target == vector(n)
+found = true;                  % Here we have found the target
+break;                         % Here we quit the search
+end
+end
+index = n;                      % index of our target
+if ~found
+index = -1;                      % If not found, the index is -1
+end
+```
+In our function, we have four inputs. The first input is the vector(n) in which we are locating our targeted value. The second input is the target that we are searching for. The third and the fourth are the first and the last value of our vector. In most cases, our first value is always one, and the last value is the entire length of the vector. But in case you want to search up to a given limit, then you input your range in the first and the last values. We also use the internal flag `found`, `true`, and `false` functions to determine whether we found our target or not.  
+
+### Example
+We have a vector M = [ 3 2 6 34 22 98 77 45 99 12] and we want to find 34. We will call our function as shown below;
+```matlab
+sequential_search(M, 34, 1, 10)
+```
+When we execute this, we get the output which is the index of our target value which is 4.
+
+![index for the target](/engineering-education/searching-and-sorting-using-matlab/search_one.png)
+
+For extensive data, it takes time for sequential search to locate a given data. This is because it compares all the data with the target value to get a similar variable. For fast searching, you need first to sort your data, but this is not the case for sequential search. Therefore, it is the only algorithm that is used to search unsorted data.
+
+### 2. Binary search
```

Comment:
Kindly explain more on binary search or link an article with more explanation. Maybe how binary search operates.
Also avoid using the word 'here' and 'it' , find an alternative that gives your article more command and style.

Do pass your article on Grammarly to help you remove the passive voices and repetitions.